### PR TITLE
backports from g-p-m

### PR DESCRIPTION
**Pull request by raveit65**
Those are all all commits from g-p-m-2.30-2.31-4
For some reasons they aren't in our code.
For a visible change you will see the primary device which is the battery on a notebook in tray-icon menu, a click on it will open m-p-m-statistics.
Guys with notebooks please test.
@clefebvre @flexiondotorg @XRevan86 @lukefromdc @monsta 

Note:
During testing on my notebook with with f24 and m-p-m from this PR or latest1.15 or 1.14 version i noticed a regression not related to this PR, as it happens with older versions too.
If notebook is on battery status the sequence **screensaver --> power off screen --> suspend** does not work.
On ac-adapter all works well.
But on rhel7 with older upower and systemd version results are different.
Here i have only a problem if screensaver is activated with using battery power.
So it would be nice if you can test this too, maybe it's hardware related here.
But again, this is not related to  this PR.

### CODE REVIEW

**Comment by raveit65** [comment]
Btw. i'm not sure about  **Add a ::devices-changed signal to GpmEngine that helper code may want**, looks like this is only for upower < 0.99.

**Comment by lukefromdc** [comment]
This is not the only place I've seen code that didn't make it to our fork of GNOME 2. Nautilus was playing around with GtkApplication stuff as far back as July 2010
https://github.com/GNOME/nautilus/commit/e5d50bfd152e3d248519160a82908470973a539f
but that never made it into MATE. Not sure if some of these commits were merged from other branches after GNOME 2.32 was released or what.  It would be useful to know what version of GNOME exactly was used in each package for the intial MATE fork, as that would tell us what should be already there.

Anything that made it into released GNOME 2.32 and hasn't been obsoleted by other library changes should be something that we could use even in GTK 2 builds.

**Comment by raveit65** [comment]
Perberos (first founder of Mate) did fork Mate from gnome2, but he didn't forked it from eg. gnome at github, in result the commit history is interrupted.
You can call it a bad fork.
I think he use used tarball for starting the git.
But nobody knows which one and i'm sure they were patched.
He left us 5 years ago, so nobody can contact him any more.

File: src/gpm-statistics.c [hunk 1]
```diff
@@ -431,7 +431,7 @@ gpm_stats_update_info_page_details (UpDevice *device)
 		/* TRANSLATORS: Present is whether the device is currently attached
 		 * to the computer, as some devices (e.g. laptop batteries) can
 		 * be removed, but still observed as devices on the system */
-		gpm_stats_add_info_data (_("Present"), gpm_stats_bool_to_string (is_present));
```
Comment (monsta) [inline src/gpm-statistics.c]: This change should be removed, or else it breaks presence status. "State" is already handled several lines below.

**Comment by monsta** [comment]
On https://github.com/mate-desktop/mate-power-manager/pull/196/commits/59c570db6c7c0e915badd9f569869fc486b4080a: we already have ```gpm_device_state_to_localised_string``` function in our code, so all this commit does is the renaming of another, unrelated function (and a define, for some reason).

**Comment by raveit65** [comment]
My goal was to back port early changes from g-p-m (gtk2) to our code.
My experiences (mate-color-manager) with code from R.Huges is to follow every step, otherwise the code is to different and it is impossible to adopt newer changes for me.
Well, i can revert the mentioned changes probably more, as others which won't work than.
But what is really wrong at current state and won't work?

**Comment by monsta** [comment]
I've corrected https://github.com/mate-desktop/mate-power-manager/pull/196/commits/42e14993cb3666515bcb9ce58691ccf3ae6d9aea a bit to make sure the warning really won't appear.

**Comment by raveit65** [comment]
Thank you.
@flexiondotorg @clefebvre 
I know you are using notebooks, can you please test the PR on bare metal?
I don't wanna merge it without testing by another person.

**Comment by monsta** [comment]
Also corrected https://github.com/mate-desktop/mate-power-manager/pull/196/commits/ceeba267e628261b694f57e56808937d37b0ac91 - removed copies of egg-dbus-monitor from both applets.

So far, it builds fine here with GTK+2.

**Comment by raveit65** [comment]
only a rebase for having distcheck: explicitly use currently selected GTK+ version in this branch.

**Comment by raveit65** [comment]
hmm, with centos7 (gtk2)
```
brightness-applet.c: In function 'gpm_applet_create_popup':
/usr/include/gtk-2.0/gtk/gtkwidget.h:471:81: error: lvalue required as left operand of assignment
 #define GTK_WIDGET_UNSET_FLAGS(wid,flag)  G_STMT_START{ (GTK_WIDGET_FLAGS (wid) &= ~(flag)); }G_STMT_END
                                                                                 ^
brightness-applet.c:713:2: note: in expansion of macro 'GTK_WIDGET_UNSET_FLAGS'
  GTK_WIDGET_UNSET_FLAGS (applet->popup, GTK_TOPLEVEL);
```

**Comment by raveit65** [comment]
with f24 gtk3
```
In file included from gpm-manager.c:60:0:
gpm-tray-icon.h:59:1: error: unknown type name 'GtkStatusIcon'
 GtkStatusIcon *gpm_tray_icon_get_status_icon  (GpmTrayIcon *icon);
 ^~~~~~~~~~~~~
gpm-manager.c:97:2: error: unknown type name 'GtkStatusIcon'
  GtkStatusIcon  *status_icon;
  ^~~~~~~~~~~~~
gpm-manager.c: In function 'gpm_manager_notify':
gpm-manager.c:487:6: warning: implicit declaration of function 'gtk_status_icon_is_embedded' [-Wimplicit-function-declaration]
      gtk_status_icon_is_embedded (manager->priv->status_icon))
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~
gpm-manager.c:488:59: warning: implicit declaration of function 'gtk_status_icon_get_icon_name' [-Wimplicit-function-declaration]
   notification = notify_notification_new (title, message, gtk_status_icon_get_icon_name(manager->priv->status_icon));
                                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gpm-manager.c:488:59: warning: passing argument 3 of 'notify_notification_new' makes pointer from integer without a cast [-Wint-conversion]
In file included from /usr/include/libnotify/notify.h:27:0,
                 from gpm-manager.c:44:
/usr/include/libnotify/notification.h:114:21: note: expected 'const char *' but argument is of type 'int'
 NotifyNotification *notify_notification_new                  (const char         *summary,
                     ^~~~~~~~~~~~~~~~~~~~~~~
Makefile:1366: recipe for target 'mate_power_manager-gpm-manager.o' failed
```
This was all working :-/

**Comment by willysr** [comment]
> For a visible change you will see the primary device which is the battery on a notebook in tray-icon menu, a click on it will open m-p-m-statistics.

This new feature works as expected. Tested in Slackware Current with MATE 1.15.1 with GTK+2

**Comment by willysr** [comment]
the screensaver pulls in normally here (i tested on AC only).
Note that i don't have systemd if that could make a difference
for me, it's +1 to merge this PR

**Comment by raveit65** [comment]
@monsta Are you fine with last commit?. Suggestion from Alexei.

**Comment by monsta** [comment]
Good catch, looks like that ```return``` was missing in GTK+3 build. Looks ok.

**Pull request merged**